components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/UserAccountRecoveryManager.java

```diff
@@ -925,30 +931,141 @@ public UserRecoveryData getUserRecoveryData(String code, RecoverySteps step) thr
         return recoveryData;
     }
 
+    /**
+     * Get user recovery data using the recovery flow id.
+     *
+     * @param recoveryFlowId Recovery flow id of the user.
+     * @param step Recovery step.
+     * @throws IdentityRecoveryException If an error occurred while validating the recoveryId.
+     */
+    public UserRecoveryData getUserRecoveryDataFromFlowId(String recoveryFlowId, RecoverySteps step)
+            throws IdentityRecoveryException {
+
+        UserRecoveryData recoveryData;
+        UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
+        try {
+            // Retrieve recovery data bound to the recoveryFlowId.
+            recoveryData = userRecoveryDataStore.loadFromRecoveryFlowId(recoveryFlowId, step);
+        } catch (IdentityRecoveryException e) {
+            // Map code expired error to new error codes for user account recovery.
+            if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_FLOW_ID.getCode().equals(e.getErrorCode())) {
+                e.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_FLOW_ID.getCode());
+            } else if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_FLOW_ID.getCode().equals(e.getErrorCode())) {
+                e.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_RECOVERY_FLOW_ID.getCode());
+            } else if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_CODE.getCode()
+                    .equals(e.getErrorCode())) {
+                e.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_EXPIRED_RECOVERY_CODE.getCode());
+            } else {
+                e.setErrorCode(Utils.prependOperationScenarioToErrorCode(e.getErrorCode(),
+                        IdentityRecoveryConstants.USER_ACCOUNT_RECOVERY));
+            }
+            throw e;
+        }
+        if (recoveryData == null) {
+            throw Utils
+                    .handleClientException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_ACCOUNT_RECOVERY_DATA,
+                            recoveryFlowId);
+        }
+        return recoveryData;
+    }
+
+    /**
+     * Get user recovery flow data using the recovery flow id.
+     *
+     * @param recoveryDataDO User Recovery Data object.
+     * @throws IdentityRecoveryException If an error occurred while validating the recoveryId.
+     */
+    public UserRecoveryFlowData loadUserRecoveryFlowData(UserRecoveryData recoveryDataDO)
```

Comment:
return param is not mentioned in the comment

Reply:
Fixed with 4c8c9f64b52364e581f4f4f1890afe0f11577c08